### PR TITLE
linter: added new `getTypeMisUse` checker

### PR DESCRIPTION
**Pull request by Danil42Russia**

### CODE REVIEW

File: src/tests/checkers/get_type_miss_use_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	if (gettype($var) != "object"){
+    }
+}
+`)
+	test.Expect = []string{
+		`use is_string instead of gettype()`,
+		`use is_float instead of gettype()`,
+		`use is_array instead of gettype()`,
+		`use is_bool instead of gettype()`,
+		`use is_object instead of gettype()`,
```
Comment (i582) [inline src/tests/checkers/get_type_miss_use_test.go:37]: Please, add cases for expression, not just variables

File: src/tests/golden/testdata/quickfix/getTypeMissUse.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    if (gettype($var) == "double"){
+    }
+
+    if (gettype($var) !== "array"){
+    }
+
+    if (gettype($var) != "boolean"){
+    }
+
+    if (gettype($var) === "object"){
```
Comment (i582) [inline src/tests/golden/testdata/quickfix/getTypeMissUse.php:16]: Please, add cases for expression, not just variables

**Review by i582 — CHANGES_REQUESTED** [review]

**Comment by YuriyNasretdinov** [comment]
I think there's one thing missing in this entire PR: why are you adding this change? Why is gettype wrong and e.g. is_string right to use?

File: src/linter/block_linter.go [hunk 3]
```diff
@@ -1039,6 +1044,69 @@ func (b *blockLinter) checkFunctionCall(e *ir.FunctionCallExpr) {
...
+	}
+
+	nodeText := b.walker.r.nodeText(firstArg)
+
+	str, ok := right.(*ir.String)
+	if !ok {
+		return
+	}
+
+	replaced, ok := phpcore.TypeToIsFunction[str.Value]
```
[the remark this answers is not in the document]
Comment (Danil42Russia) [reply src/linter/block_linter.go]: done

File: src/tests/golden/testdata/quickfix/getTypeMissUse.php (hunk 2 above)
Comment (Danil42Russia) [reply src/tests/golden/testdata/quickfix/getTypeMissUse.php]: done

File: src/tests/checkers/get_type_miss_use_test.go (hunk 1 above)
Comment (Danil42Russia) [reply src/tests/checkers/get_type_miss_use_test.go:43]: done

File: src/linter/quickfix.go [hunk 4]
```diff
@@ -46,3 +46,19 @@ func (g *QuickFixGenerator) NullForNotNullableProperty(prop *ir.PropertyStmt) qu
 		Replacement: string(withoutAssign),
 	}
 }
+
+func (g *QuickFixGenerator) GetType(node ir.Node, replaced, nodeText string, isNegative bool) quickfix.TextEdit {
```
[the remark this answers is not in the document]
Comment (Danil42Russia) [reply src/linter/quickfix.go]: done

**Pull request merged**